incubator/virtualcluster/pkg/syncer/resources/pod/dws.go

```diff
@@ -45,55 +45,59 @@ func (c *controller) StartDWS(stopCh <-chan struct{}) error {
 	return c.multiClusterPodController.Start(stopCh)
 }
 
-func (c *controller) Reconcile(request reconciler.Request) (reconciler.Result, error) {
+func (c *controller) Reconcile(request reconciler.Request) (res reconciler.Result, retErr error) {
 	klog.V(4).Infof("reconcile pod %s/%s for cluster %s", request.Namespace, request.Name, request.ClusterName)
 	targetNamespace := conversion.ToSuperMasterNamespace(request.ClusterName, request.Namespace)
+
 	pPod, err := c.podLister.Pods(targetNamespace).Get(request.Name)
-	pExists := true
-	if err != nil {
-		if !errors.IsNotFound(err) {
-			return reconciler.Result{Requeue: true}, err
-		}
-		pExists = false
+	if err != nil && !errors.IsNotFound(err) {
+		return reconciler.Result{Requeue: true}, err
 	}
-	vExists := true
+
+	var vPod *v1.Pod
 	vPodObj, err := c.multiClusterPodController.Get(request.ClusterName, request.Namespace, request.Name)
-	if err != nil {
-		if !errors.IsNotFound(err) {
-			return reconciler.Result{Requeue: true}, err
-		}
-		vExists = false
+	if err == nil {
+		vPod = vPodObj.(*v1.Pod)
+	} else if !errors.IsNotFound(err) {
+		return reconciler.Result{Requeue: true}, err
 	}
 
 	var operation string
-	if vExists && !pExists {
+	defer func() {
+		recordOperationDuration(operation, time.Now())
+		recordOperationStatus(operation, retErr)
+	}()
+
+	if vPod != nil && pPod == nil {
 		operation = "pod_add"
-		defer recordOperationDuration(operation, time.Now())
-		vPod := vPodObj.(*v1.Pod)
 		err := c.reconcilePodCreate(request.ClusterName, targetNamespace, request.UID, vPod)
-		recordOperationStatus(operation, err)
 		if err != nil {
 			klog.Errorf("failed reconcile Pod %s/%s CREATE of cluster %s %v", request.Namespace, request.Name, request.ClusterName, err)
+			go func() {
+				if parentRef := getParentRefFromPod(vPod); parentRef != nil {
+					c.multiClusterPodController.Eventf(request.ClusterName, parentRef, v1.EventTypeWarning, "FailedCreate", "Error creating: %v", err)
+				}
+				c.multiClusterPodController.Eventf(request.ClusterName, &v1.ObjectReference{
+					Kind:      "Pod",
+					Namespace: vPod.Namespace,
+					UID:       vPod.UID,
+				}, v1.EventTypeWarning, "FailedCreate", "Error creating: %v", err)
```

Comment:
we filled the `controller` field in event, so unnecessary to contains context info in error message